File: CANOAS MALUCAS/C#/CanoasMalucas/CanoasMalucas/Canoa.cs
using System;
using System.Collections;
using System.Collections.Generic;

public class Canoa
{
    public string nome;
    public bool[] blocos;

    public Canoa(int tamanho)
    {
        blocos = new bool[tamanho];
        InicializarBlocos();
    }

    public void InicializarBlocos()
    {
        for (int indice = 0; indice < blocos.Length; indice++)
        {
            blocos[indice] = true;
        }
    }

    public void RecontruirBloco(int posicao)
    {
        if (posicao >= 0 && posicao < blocos.Length)
        {
            blocos[posicao] = true;
        }
    }

    public void DestruirBloco(int posicao)
    {
        if (posicao >= 0 && posicao < blocos.Length)
        {
            blocos[posicao] = false;
        }
    }

    public void Afundar()
    {
        for (int indice = 0; indice < blocos.Length; indice++)
        {
            blocos[indice] = false;
        }
    }

    public override string ToString()
    {
        string textBlocos = "";

        foreach (bool bloco in blocos)
        {
            textBlocos += "[" + bloco + "] ";
        }

        return "Canoa{\n" +
                "  + nome = " + nome + "\n" +
                "  + blocos = " + textBlocos + "\n" +
                "}";
    }

}

File: CANOAS MALUCAS/JAVA/CanoasMalucas3_0/src/Canoa.java
public class Canoa{
    public String nome = "";
    public boolean[] blocos;
    public Canoa(int tamanho){
        blocos = new boolean[tamanho];
        inicializarBlocos();
    }
    public void inicializarBlocos(){
        for (int indice = 0; indice < blocos.length; indice++) {
            blocos[indice] = true;
        }
    }
    public void recontruirBloco(int posicao){
        if (posicao >= 0 && posicao < blocos.length) {
            blocos[posicao] = true;
        }
    }
    public void destruirBloco(int posicao){
        if (posicao >= 0 && posicao < blocos.length) {
            blocos[posicao] = false;
        }
    }
    public void afundar(){
        for (int indice = 0; indice < blocos.length; indice++) {
            blocos[indice] = false;
        }
    }
    @Override
    public String toString() {
        String textBlocos = "";
        for (boolean bloco : blocos) {
            textBlocos += "[" + bloco + "] ";
        }        
        return  "Canoa{\n" + 
                "  + nome = " + nome + "\n" + 
                "  + blocos = " + textBlocos + "\n" + 
                "}";
    }
}